app/Http/Requests/TourDeepDiveEmailRequest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+use App\Tour;
+
+class TourDeepDiveEmailRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            'email' => ['required', 'email'],
+            'deepdiveIds' => ['required', 'array'],
+            'deepdiveIds.*' => 'uuid',
+            'locale' => [
+                'required',
+                Rule::in(Tour::possibleLocales())
```

Comment:
Probably not worth undoing now, but if this is the only place we need to know locale server side, I'd probably have voted to not bother adding all this - my philosophy is that if a form we control is passing us an invalid value here, something is probably horrifically wrong so whether or not we validate it doesn't really matter. 

Not a criticism and obviously we may as well keep it, just sharing some of my thinking on values on stuff like this versus the (time) cost and added complexity.

Reply:
I see that. tbh, I've been hearing the nice youtube laravel guy talking about custom form requests and validation, and wanted to give it a whirl.